macros/get_metric_sql.sql

```diff
@@ -41,6 +41,14 @@ within the final dataset in order to accomplish base + secondary calc functional
 a custom calendar -#}
 {%- set calendar_tbl = ref(var('dbt_metrics_calendar_model', "dbt_metrics_default_calendar")) -%}
 
+{#- Reducing the metric dictionary to just metric configs that are relevant for secondary calcs
+    right now the metric dictionary only gets passed to the build_metric_sql macro, which has config information already
+-#}
+{%- set metric_config_dict = {} -%}
```

Comment:
@dave-connors-3 can you give me the ELI5 why we're creating a new dict here and not just passing along the metric_dictionary to anywhere that needs this info? 

Background: The reason we created the metric dictionary is because of develop. In the original version of the package we could just pass the metric object everywhere and it was easy peasy. With develop we had to come up with an alternative so that was the dictionary - a sort of proxy metric object. I'm curious to the advantages that we gain by creating a new variable for config information instead of having it live with the metric in the same dict

Reply:
I'm not personally convinced there are any benefits 😅 

Configs are kinda funny to me because they in the grey area between a core component of a metric (which would be a top level metric parameter!) and the rubber meets the road of making a metric come to life. 

It felt more correct to pass the "how to I make this metric look" type of objects around instead of the entire "what is this metric" to macros that control a very small piece of the puzzle. I.e. why would `gen_secondary_metrics_cte` need to know about the available grains, or the model the metric relies on? 

That being said, it's 100% redundant to put the config then immediately pull it out of the metric dict, so the solution doesn't even really feel better than the issue.

Reply:
I'm not particularly bothered by passing around the entire "what is this metric" object because it mimics the pattern we would do if we didn't need to convert it to a dict. In that world we'd probably reference `metric.calc_config` instead of this big dict of dicts. My advocacy would be to keep it in the metric dict and then pass that.

Reply:
i am on board